Dockerfile

```diff
@@ -0,0 +1,30 @@
+FROM python:3.12.0-slim
+
+ENV PYTHONUNBUFFERED=1
+
+WORKDIR /app
+
+ARG UID=10001
+RUN adduser \
+    --disabled-password \
+    --gecos "" \
+    --home "/app" \
+    --shell "/sbin/nologin" \
+    --no-create-home \
+    --uid "${UID}" \
+    django-user
+
+# Install dependencies
+COPY requirements.txt .
+RUN pip install --upgrade pip && \
+    pip install --no-cache-dir -r requirements.txt
+
+# Copy application files and ensure correct ownership
+COPY . .
```

Comment:
Consider using a more specific COPY command to avoid copying unnecessary files into the Docker image. This can help reduce the image size and improve build performance. You can specify only the necessary directories or files needed for the application.